### PR TITLE
Gather subscriptions from clusters

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

**Comment by dmage** [comment]
/assign @wking 

These resources are useful for jobs that install third-party operators.

File: ci-operator/step-registry/gather/extra/gather-extra-commands.sh [hunk 1]
```diff
@@ -95,9 +95,13 @@ queue ${ARTIFACT_DIR}/oc_cmds/roles oc --insecure-skip-tls-verify --request-time
...
 queue ${ARTIFACT_DIR}/oc_cmds/services oc --insecure-skip-tls-verify --request-timeout=5s get services --all-namespaces
 FILTER=gzip queue ${ARTIFACT_DIR}/statefulsets.json.gz oc --insecure-skip-tls-verify --request-timeout=5s get statefulsets --all-namespaces -o json
-queue ${ARTIFACT_DIR}/statefulsets oc --insecure-skip-tls-verify --request-timeout=5s get statefulsets --all-namespaces
+queue ${ARTIFACT_DIR}/oc_cmds/statefulsets oc --insecure-skip-tls-verify --request-timeout=5s get statefulsets --all-namespaces
 queue ${ARTIFACT_DIR}/routes.json oc --insecure-skip-tls-verify --request-timeout=5s get routes --all-namespaces -o json
 queue ${ARTIFACT_DIR}/oc_cmds/routes oc --insecure-skip-tls-verify --request-timeout=5s get routes --all-namespaces
+queue ${ARTIFACT_DIR}/subscriptions.json oc --insecure-skip-tls-verify --request-timeout=5s get subscriptions --all-namespaces -o json
+queue ${ARTIFACT_DIR}/oc_cmds/subscriptions oc --insecure-skip-tls-verify --request-timeout=5s get subscriptions --all-namespaces
+queue ${ARTIFACT_DIR}/clusterserviceversions.json oc --insecure-skip-tls-verify --request-timeout=5s get clusterserviceversions --all-namespaces -o json
```
Comment (vrutkovs) [inline ci-operator/step-registry/gather/extra/gather-extra-commands.sh:103]:
```suggestion
```
ditto

**Review by vrutkovs — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/gather/extra/gather-extra-commands.sh [hunk 2]
```diff
@@ -95,9 +95,13 @@ queue ${ARTIFACT_DIR}/oc_cmds/roles oc --insecure-skip-tls-verify --request-time
 queue ${ARTIFACT_DIR}/services.json oc --insecure-skip-tls-verify --request-timeout=5s get services --all-namespaces -o json
 queue ${ARTIFACT_DIR}/oc_cmds/services oc --insecure-skip-tls-verify --request-timeout=5s get services --all-namespaces
 FILTER=gzip queue ${ARTIFACT_DIR}/statefulsets.json.gz oc --insecure-skip-tls-verify --request-timeout=5s get statefulsets --all-namespaces -o json
-queue ${ARTIFACT_DIR}/statefulsets oc --insecure-skip-tls-verify --request-timeout=5s get statefulsets --all-namespaces
+queue ${ARTIFACT_DIR}/oc_cmds/statefulsets oc --insecure-skip-tls-verify --request-timeout=5s get statefulsets --all-namespaces
```
[the remark this answers is not in the document]
Comment (dmage) [reply ci-operator/step-registry/gather/extra/gather-extra-commands.sh:98]: The pattern that I see in this file is that everything is collected twice:

`/XXX.json` is output with `-o json`
`/oc_cmds/XXX` is output of the same command, but without `-o` at all or with `-o wide`

`statefulsets` is the only exception, they are collected twice but have different file names.

File: ci-operator/step-registry/gather/extra/gather-extra-commands.sh [hunk 3]
```diff
@@ -95,9 +95,13 @@ queue ${ARTIFACT_DIR}/oc_cmds/roles oc --insecure-skip-tls-verify --request-time
...
 queue ${ARTIFACT_DIR}/routes.json oc --insecure-skip-tls-verify --request-timeout=5s get routes --all-namespaces -o json
 queue ${ARTIFACT_DIR}/oc_cmds/routes oc --insecure-skip-tls-verify --request-timeout=5s get routes --all-namespaces
+queue ${ARTIFACT_DIR}/subscriptions.json oc --insecure-skip-tls-verify --request-timeout=5s get subscriptions --all-namespaces -o json
```
[the remark this answers is not in the document]
Comment (vrutkovs) [reply ci-operator/step-registry/gather/extra/gather-extra-commands.sh:101]: Hmm, lets follow the same pattern until we're sure human-readable is no longer used

**Review by vrutkovs — APPROVED** [review]
/lgtm

**Pull request merged**